### ISSUE TITLE
storage_proxy_replica counters seem incorrect - some of them have  negative valued / others are not incremenented

### ISSUE THREAD

**Issue by slivne** [scylladb/scylla#1988]
Version: 1.6

some metrics seem incorrect

Please note this is via the collectd interface not the prometheus interface

not sure if its related to #1950 - it seems unrelated

```
collectd_storage_proxy_replica_total_operations{exported_instance="ip-172-30-0-197",instance="52.87.229.55:9103",job="scylla",storage_proxy_replica="2",type="received_mutations"}	139774817954816
collectd_storage_proxy_replica_total_operations{exported_instance="ip-172-30-0-40",instance="54.172.16.148:9103",job="scylla",storage_proxy_replica="3",type="forwarded_mutations"}	140284029529088
collectd_storage_proxy_replica_total_operations{exported_instance="ip-172-30-0-48",instance="54.166.35.25:9103",job="scylla",storage_proxy_replica="3",type="received_mutations"}	-4827825075205571000
collectd_storage_proxy_replica_total_operations{exported_instance="ip-172-30-0-246",instance="54.165.150.207:9103",job="scylla",storage_proxy_replica="0",type="received_mutations"}	3697259913025704000
collectd_storage_proxy_replica_total_operations{exported_instance="ip-172-30-0-229",instance="54.152.215.132:9103",job="scylla",storage_proxy_replica="0",type="forwarded_mutations"}	-5723243642092800000
collectd_storage_proxy_replica_total_operations{exported_instance="ip-172-30-0-40",instance="54.172.16.148:9103",job="scylla",storage_proxy_replica="3",type="received_mutations"}	140284029529088
collectd_storage_proxy_replica_total_operations{exported_instance="ip-172-30-0-248",instance="54.172.201.161:9103",job="scylla",storage_proxy_replica="11",type="forwarding_errors"}	3591772330067
collectd_storage_proxy_replica_total_operations{exported_instance="ip-172-30-0-248",instance="54.172.201.161:9103",job="scylla",storage_proxy_replica="10",type="forwarded_mutations"}	3591772330067
collectd_storage_proxy_replica_total_operations{exported_instance="ip-172-30-0-229",instance="54.152.215.132:9103",job="scylla",storage_proxy_replica="3",type="forwarded_mutations"}	-4827825074771526000
collectd_storage_proxy_replica_total_operations{exported_instance="ip-172-30-0-248",instance="54.172.201.161:9103",job="scylla",storage_proxy_replica="11",type="received_mutations"}	3591772330067
collectd_storage_proxy_replica_total_operations{exported_instance="ip-172-30-0-197",instance="52.87.229.55:9103",job="scylla",storage_proxy_replica="0",type="received_mutations"}	105553205303809
collectd_storage_proxy_replica_total_operations{exported_instance="ip-172-30-0-48",instance="54.166.35.25:9103",job="scylla",storage_proxy_replica="6",type="forwarding_errors"}	140408203996160
collectd_storage_proxy_replica_total_operations{exported_instance="ip-172-30-0-40",instance="54.172.16.148:9103",job="scylla",storage_proxy_replica="0",type="forwarded_mutations"}	1202797601056961300
collectd_storage_proxy_replica_total_operations{exported_instance="ip-172-30-0-197",instance="52.87.229.55:9103",job="scylla",storage_proxy_replica="0",type="forwarded_mutations"}	105553205303809
collectd_storage_proxy_replica_total_operations{exported_instance="ip-172-30-0-48",instance="54.166.35.25:9103",job="scylla",storage_proxy_replica="6",type="forwarded_mutations"}	140408203996160
collectd_storage_proxy_replica_total_operations{exported_instance="ip-172-30-0-229",instance="54.152.215.132:9103",job="scylla",storage_proxy_replica="3",type="forwarding_errors"}	-4827825074771526000
collectd_storage_proxy_replica_total_operations{exported_instance="ip-172-30-0-246",instance="54.165.150.207:9103",job="scylla",storage_proxy_replica="3",type="forwarded_mutations"}	105759296795760
collectd_storage_proxy_replica_total_operations{exported_instance="ip-172-30-0-40",instance="54.172.16.148:9103",job="scylla",storage_proxy_replica="10",type="forwarded_mutations"}	140283985488896
collectd_storage_proxy_replica_total_operations{exported_instance="ip-172-30-0-246",instance="54.165.150.207:9103",job="scylla",storage_proxy_replica="3",type="received_mutations"}	105759296795760
collectd_storage_proxy_replica_total_operations{exported_instance="ip-172-30-0-246",instance="54.165.150.207:9103",job="scylla",storage_proxy_replica="9",type="forwarding_errors"}	140721020962208
collectd_storage_proxy_replica_total_operations{exported_instance="ip-172-30-0-40",instance="54.172.16.148:9103",job="scylla",storage_proxy_replica="10",type="received_mutations"}	140283985488896
collectd_storage_proxy_replica_total_operations{exported_instance="ip-172-30-0-229",instance="54.152.215.132:9103",job="scylla",storage_proxy_replica="0",type="received_mutations"}	-5723243642092800000
collectd_storage_proxy_replica_total_operations{exported_instance="ip-172-30-0-40",instance="54.172.16.148:9103",job="scylla",storage_proxy_replica="4",type="forwarded_mutations"}	140284023237632
collectd_storage_proxy_replica_total_operations{exported_instance="ip-172-30-0-229",instance="54.152.215.132:9103",job="scylla",storage_proxy_replica="3",type="received_mutations"}	-4827825074771526000
collectd_storage_proxy_replica_total_operations{exported_instance="ip-172-30-0-246",instance="54.165.150.207:9103",job="scylla",storage_proxy_replica="0",type="forwarding_errors"}	3697259913025704000
collectd_storage_proxy_replica_total_operations{exported_instance="ip-172-30-0-40",instance="54.172.16.148:9103",job="scylla",storage_proxy_replica="3",type="forwarding_errors"}	140284029529088
collectd_storage_proxy_replica_total_operations{exported_instance="ip-172-30-0-48",instance="54.166.35.25:9103",job="scylla",storage_proxy_replica="13",type="forwarded_mutations"}	140408159955968
```

**Comment by gleb-cloudius** [comment]
On Tue, Jan 03, 2017 at 05:36:40AM -0800, Shlomi Livne wrote:
> Version: 1.6
> 
> some metrics seem incorrect
> 
They all look like garbage.

--
			Gleb.

**Comment by vladzcloudius** [comment]
Yes. This is related to the d227f531f911e5b7f9f45612ac4b8dcbb5944770 seastar patch. We should either capture the values in lambda or backport the above patch to 1.6.

**Comment by slivne** [comment]
fixed by c6928247863964e0147ffaf9391d0837f82b3a98
